nifi-nar-bundles/nifi-jolt-record-bundle/nifi-jolt-record-processors/src/main/java/org/apache/nifi/processors/jolt/record/JoltTransformRecord.java

```diff
@@ -296,39 +297,37 @@ public void onTrigger(final ProcessContext context, ProcessSession session) thro
+            // Record Writer so that if the Record Writer chooses to inherit the Record Schema from the Record itself, it will inherit the transformed schema, not the
+            // schema determined by the Record Reader.
+            final Record firstRecord = reader.nextRecord();
+            if (firstRecord == null) {
+                try (final OutputStream out = session.write(transformed);
+                     final RecordSetWriter writer = writerFactory.createWriter(getLogger(), schema, out, transformed)) {
 
-            try {
-                // We want to transform the first record before creating the Record Writer. We do this because the Record will likely end up with a different structure
-                // and therefore a difference Schema after being transformed. As a result, we want to transform the Record and then provide the transformed schema to the
-                // Record Writer so that if the Record Writer chooses to inherit the Record Schema from the Record itself, it will inherit the transformed schema, not the
-                // schema determined by the Record Reader.
-                final Record firstRecord = reader.nextRecord();
-                if (firstRecord == null) {
-                    try (final OutputStream out = session.write(transformed);
-                         final RecordSetWriter writer = writerFactory.createWriter(getLogger(), schema, out, transformed)) {
-
-                        writer.beginRecordSet();
-                        writeResult = writer.finishRecordSet();
-
-                        attributes.put("record.count", String.valueOf(writeResult.getRecordCount()));
-                        attributes.put(CoreAttributes.MIME_TYPE.key(), writer.getMimeType());
-                        attributes.putAll(writeResult.getAttributes());
-                    }
+                    writer.beginRecordSet();
+                    writeResult = writer.finishRecordSet();
 
-                    transformed = session.putAllAttributes(transformed, attributes);
-                    session.transfer(transformed, REL_SUCCESS);
-                    session.transfer(original, REL_ORIGINAL);
-                    logger.info("{} had no Records to transform", new Object[]{original});
-                    return;
+                    attributes.put("record.count", String.valueOf(writeResult.getRecordCount()));
+                    attributes.put(CoreAttributes.MIME_TYPE.key(), writer.getMimeType());
+                    attributes.putAll(writeResult.getAttributes());
                 }
 
+                transformed = session.putAllAttributes(transformed, attributes);
+                logger.info("{} had no Records to transform", new Object[]{original});
+            } else {
+
                 final JoltTransform transform = getTransform(context, original);
                 final Record transformedFirstRecord = transform(firstRecord, transform);
```

Comment:
If the transform fails here transformedFirstRecord is returned and causes a NullPointerException on the next line for writeSchema.  Recommend adding a null check.